lib/fe/review.ex

```diff
@@ -0,0 +1,241 @@
+defmodule FE.Review do
+  @moduledoc """
+  `FE.Review` is a data type similar to `FE.Result`, made for representing
+  output of a computation that either succeed (`accepted`) or fail (`rejected`),
+  but that might continue despite of issues encountered (`issues`).
+
+  One could say that the type is a specific implementation of a writer monad,
+  that collects issues encountered during some computation.
+
+  For instance, it might be used for validation of a user input, when we don't
+  want to stop the process of validation when we encounter the first mistake,
+  but rather we would like to collect all the user's mistakes before returning
+  feedback to her.
+  """
+
+  @type t(a, b) :: {:accepted, a} | {:issues, a, [b]} | {:rejected, [b]}
```

Comment:
I'm wondering if `rejected` variant is really necessary here. After all, when reviewing stuff, there always is an object under review. On the other hand, having rejected allows halting the chain of `and_then`'s sooner or prevents us from mapping the value if we really only care about issues.

Thoughts?